beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -72,17 +72,29 @@ func setExecutionData(ctx context.Context, blk interfaces.SignedBeaconBlock, loc
 		return local.Bid, local.BlobsBundle, setLocalExecution(blk, local)
 	}
 
-	var builderKzgCommitments [][]byte
 	builderPayload, err := bid.Header()
 	if err != nil {
 		log.WithError(err).Warn("Proposer: failed to retrieve header from BuilderBid")
 		return local.Bid, local.BlobsBundle, setLocalExecution(blk, local)
 	}
-	//TODO: add builder execution requests here.
+
+	var builderKzgCommitments [][]byte
 	if bid.Version() >= version.Deneb {
-		builderKzgCommitments, err = bid.BlobKzgCommitments()
-		if err != nil {
-			log.WithError(err).Warn("Proposer: failed to retrieve kzg commitments from BuilderBid")
+		bidDeneb, ok := bid.(builder.BidDeneb)
+		if !ok {
+			log.Warnf("bid type %T does not implement builder.BidDeneb ", bid)
+		} else {
+			builderKzgCommitments = bidDeneb.BlobKzgCommitments()
+		}
+	}
+
+	var executionRequests *enginev1.ExecutionRequests
+	if bid.Version() >= version.Electra {
+		bidElectra, ok := bid.(builder.BidElectra)
+		if !ok {
+			log.Warnf("bid type %T does not implement builder.BidElectra ", bid)
+		} else {
+			executionRequests = bidElectra.ExecutionRequests()
```

Comment:
we dont need to do this until after: `higherValueBuilder && withdrawalsMatched`
for the default bellatrix case, we can just set them as nil: `setBuilderExecution(blk, builderPayload, nil, nil)`